test/libsolidity/SemanticTest.cpp

```diff
@@ -78,15 +78,35 @@ TestCase::TestResult SemanticTest::run(ostream& _stream, string const& _linePref
 		for (auto& test: m_tests)
 			test.reset();
 
+		map<string, dev::test::Address> libraries;
+
+		bool constructed = false;
+
 		for (auto& test: m_tests)
 		{
-			if (&test == &m_tests.front())
-				if (test.call().isConstructor)
-					deploy("", test.call().value, test.call().arguments.rawBytes());
+			if (!constructed)
+				if (test.call().isLibrary)
+				{
+					soltestAssert(
+						deploy(test.call().signature, 0, {}, libraries) && m_transactionSuccessful,
+						"Failed to deploy library " + test.call().signature
+					);
+					libraries[test.call().signature] = m_contractAddress;
+					continue;
+				}
 				else
-					soltestAssert(deploy("", 0, bytes()), "Failed to deploy contract.");
+				{
+					if (test.call().isConstructor)
+						deploy("", test.call().value, test.call().arguments.rawBytes(), libraries);
```

Comment:
Maybe not for this PR, but one could also sort a copy of `m_tests` before such that libraries would be deployed first

Reply:
Yeah - or parse libraries and constructors separately from the actual calls... but I'd postpone that.